src/runtime/thread_pool.cc

```diff
@@ -310,9 +352,33 @@ class ThreadPool {
       }
     }
   }
+  // bind worker threads to disjoint cores
+  void SetThreadAffinity() {
+#if defined(__ANDROID__)
+  #define CPU_SETSIZE 1024
+  #define __NCPUBITS (8 * sizeof (uint64_t))
+  typedef struct {
+    uint64_t __bits[CPU_SETSIZE / __NCPUBITS];
+  } cpu_set_t;
+
+  #define CPU_SET(cpu, cpusetp) \
+    ((cpusetp)->__bits[(cpu)/__NCPUBITS] |= (1UL << ((cpu) % __NCPUBITS)))
+  #define CPU_ZERO(cpusetp) \
+    memset((cpusetp), 0, sizeof(cpu_set_t))
+#endif
+    for (int i=0; i < num_workers_; ++i) {
+#if defined(__linux__) || defined(__ANDROID__)
+      cpu_set_t cpuset;
+      CPU_ZERO(&cpuset);
+      CPU_SET(i, &cpuset);
+      pthread_setaffinity_np(threads_[i].native_handle(),
```

Comment:
pthread_setaffinity_np: Pthread support for android is limited. This API doesn't work.
Now android native compilation give an error here.

Reply:
I do not have android device at hand to test out. Contribution on this is encouraged. Thanks!

Reply:
PR for android https://github.com/dmlc/tvm/pull/986